publisher-sdk/src/androidTest/java/com/criteo/publisher/logging/PublisherCodeRemoverTest.kt

```diff
@@ -156,6 +156,29 @@ class PublisherCodeRemoverTest {
         .hasStackTraceContainingOnlyAllowedStack()
   }
 
+  @Test
+  fun removePublisherCode_GivenSdkExceptionWithMessageOfPublisherException_RemovePublisherMessageInSdkException() {
+    val publisherCode = DummyPublisherCode.sdkDummyInterfaceThrowingGenericException()
+
+    // Stack is: JUnit + this test (intercept exception) + publisher code (cause)
+    val exception = exceptionThrownBy {
+      try {
+        publisherCode.foo()
+      } catch (e: Exception) {
+        throw SdkException(SdkException(SdkException(e))) // JDK copy the publisher's message into the SDK exception
```

Comment:
Why is the `SdkException` wrapped twice in an `SdkException`?

Reply:
This is to demonstrate that solution works recursively and works on exception caused by publishers but also on exception caused by the SDK (which is caused by publishers), ...:
- let's say the message of e is `foo` and e is of type `FooException`
- `SdkException(e).message` is then `FooException: foo`
- `SdkException(SdkException(e)).message` is then `SdkException: FooException: foo`
- `SdkException(SdkException(SdkException(e))).message` is then `SdkException: SdkException: FooException: foo`

Expected messages are, respectively:
- `An exception occurred from publisher's code` (with exception of type `PublisherException`)
- `PublisherException: An exception occurred from publisher's code`
- `SdkException: PublisherException: An exception occurred from publisher's code`
- `SdkException: SdkException: PublisherException: An exception occurred from publisher's code`

Reply:
Ok thanks for the explanation. I think it's really worth documenting that in the code.

Reply:
Approving the PR anyway. This can be done in a separate PR.

Reply:
Done